corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -287,6 +305,35 @@ def rebuild_from_transactions(self, transactions, rebuild_transaction, unarchive
         if not self.case.modified_on:
             self.case.modified_on = rebuild_transaction.server_date
 
+    def reconcile_transactions_if_necessary(self):
+        if self.case.check_transaction_order():
+            return False
+        datadog_counter("commcare.form_processor.sql.reconciling_transactions")
+        try:
+            self.reconcile_transactions()
+        except ReconciliationError as e:
+            reconciliation_soft_assert(False, "ReconciliationError: %s" % e.message)
+        finally:
+            return True
```

Comment:
nit? `finally` not necessary here. Not even sure what it means to have a bare `return` statement in a finally block—I guess nothing special. It will certainly not `return True` if an exception is raised, although the semantics remind me of `On Error Resume Next` from Visual Basic (shudder).

Reply:
e74e36d